dm/syncer/syncer.go

```diff
@@ -1621,11 +1632,13 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 
 			if s.streamerController.CanRetry(err) {
 				// lastLocation is the last finished GTID
-				err = s.streamerController.ResetReplicationSyncer(tctx, lastLocation)
+				err = s.streamerController.ResetReplicationSyncer(tctx, s.checkpoint.GlobalPoint())
 				if err != nil {
 					return err
 				}
 				log.L().Info("reset replication binlog puller", zap.Any("pos", s.checkpoint.GlobalPoint()))
+				_ = s.safeMode.Add(tctx, 1)
```

Comment:
How about use a local variable like `inFinerRetry` to record this temporary safe-mode
```
safeMode: s.safeMode.Enable() | retrySafeMode,
```